RestaurantReservation.API/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,166 @@
+
+            return Ok(_mapper.Map<EmployeeWithoutDetailsDto>(employee));
+        }
+
+        /// <summary>
+        /// Create new employee
+        /// </summary>
+        /// <param name="employeeDto">Employee data</param>
+        /// <returns>The created employee</returns>
+        /// <response code="201">Returns the created employee</response>
+        [HttpPost]
+        public async Task<ActionResult<EmployeeDto>> Create(EmployeeDto employeeDto)
+        {
+            var employeeToAdd = _mapper.Map<Employee>(employeeDto);
+            await _employeeRepository.AddAsync(employeeToAdd);
+            var employeeToReturn = _mapper.Map<EmployeeWithoutDetailsDto>(employeeToAdd);
+            return CreatedAtRoute("GetEmployee", new { id = employeeToReturn.Id }, employeeToReturn);
+        }
+
+        /// <summary>
+        /// Delete employee by Id
+        /// </summary>
+        /// <param name="id">Id of employee to delete</param>
+        /// <returns>No content if success</returns>
+        /// <response code="204">If employee deleted successfully</response>
+        /// <response code="404">If no employee with the specified id is found</response>
+        [HttpDelete("{id}")]
+        public async Task<ActionResult> Delete(int id)
+        {
+            var existingEmployee = await _employeeRepository.GetByIdAsync(id);
+            if (existingEmployee == null)
+            {
+                var error = EmployeeErrors.NotFound(id);
+                return NotFound(error);
+            }
+            await _employeeRepository.DeleteAsync(existingEmployee);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Update employee by Id
+        /// </summary>
+        /// <param name="id">Id of employee to update</param>
+        /// <param name="employee">The updated employee data</param>
+        /// <returns>No content if success</returns>
+        /// <response code="204">If employee updated successfully</response>
+        /// <response code="404">If no employee with the specified id is found</response>
+        [HttpPut("{id}")]
+        public async Task<ActionResult> Update(int id, EmployeeDto employee)
+        {
+            var existingEmployee = await _employeeRepository.GetByIdAsync(id);
+            if (existingEmployee == null)
+            {
+                var error = EmployeeErrors.NotFound(id);
+                return NotFound(error);
+            }
+            _mapper.Map(employee, existingEmployee);
+            await _employeeRepository.UpdateAsync(existingEmployee);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Get managers
+        /// </summary>
+        /// <param name="pageNumber">The page number to return</param>
+        /// <param name="pageSize">Max number of records per page</param>
+        /// <returns>List of managers</returns>
+        /// <response code="200">Returns the list of managers</response>
+        [HttpGet("managers")]
```

Comment:
api/Employees/Managers  - mmm I don't think this adhear to rest, maybe we can just give the =getEmployee action the ability to filter on the role so the route would be api/employees?role=manager